src/main/java/de/qabel/desktop/config/RepositoryBasedClientConfig.java

```diff
@@ -0,0 +1,174 @@
+        } catch (PersistenceException | EntityNotFoundExcepion e) {
+            throw new IllegalStateException("couldn't load current account", e);
+        }
+    }
+
+    @Override
+    public void setAccount(Account account) throws IllegalStateException {
+        try {
+            accountRepo.save(account);
+            configRepo.save("account", String.valueOf(account.getId()));
+            accountListener.forEach(c -> c.accept(account));
+        } catch (PersistenceException e) {
+            throw new IllegalStateException("failed to set account", e);
+        }
+    }
+
+    @Override
+    public void onSetAccount(Consumer<Account> consumer) {
+        accountListener.add(consumer);
+    }
+
+    @Override
+    public Identity getSelectedIdentity() {
+        try {
+            if (!configRepo.contains("identity") || configRepo.find("identity") == null) {
+                return null;
+            }
+            return identityRepo.find(Integer.valueOf(configRepo.find("identity")));
+        } catch (PersistenceException | EntityNotFoundExcepion e) {
+            throw new IllegalStateException("failed to load identity", e);
+        }
+    }
+
+    @Override
+    public void selectIdentity(Identity identity) {
+        try {
+            if (identity == null) {
+                configRepo.save("identity", null);
+            } else {
+                identityRepo.save(identity);
+                configRepo.save("identity", String.valueOf(identity.getId()));
+            }
+            identityListener.forEach(c -> c.accept(identity));
+        } catch (PersistenceException e) {
+            throw new IllegalStateException("failed to select identity", e);
+        }
+    }
+
+    @Override
+    public void onSelectIdentity(Consumer<Identity> consumer) {
+        identityListener.add(consumer);
+    }
+
+    @Override
+    public boolean hasDeviceId() {
+        try {
+            return configRepo.contains("device_id");
+        } catch (PersistenceException e) {
+            throw new IllegalStateException("failed to check device id", e);
+        }
+    }
+
+    @Override
+    public void setDeviceId(String deviceId) {
+        try {
+            configRepo.save("device_id", deviceId);
+        } catch (PersistenceException e) {
+            throw new IllegalStateException("failed to set device id", e);
+        }
+    }
+
+    @Override
+    public String getDeviceId() {
+        try {
+            return configRepo.find("device_id");
+        } catch (PersistenceException | EntityNotFoundExcepion e) {
+            throw new IllegalStateException("failed to check device id", e);
+        }
+    }
+
+    @Override
+    public Date getLastDropPoll(Identity identity) {
+        String drop = getDrop(identity);
+        return getLastDropPoll(drop);
+    }
+
+    public String getDrop(Identity identity) {
+        Set<DropURL> drops = identity.getDropUrls();
+        return drops.toArray(new DropURL[drops.size()])[0].toString();
```

Comment:
So this returns the drop url for an Identity, but the method is called getDrop and not getDropURL and it returns a string instead of a DropURL